### ISSUE TITLE
IS NULL / IS NOT NULL reported as anti-pattern

### ISSUE THREAD

**Issue by vfaronov** [jarulraj/sqlcheck#3]
```
$ ./sqlcheck 
-------------------------------------------------
> RISK LEVEL    :: ALL ANTI-PATTERNS
-------------------------------------------------
==================== Results ===================
SELECT foo FROM bar WHERE baz IS NULL;

-------------------------------------------------
SQL Statement: select foo from bar where baz is null;
(LOW RISK) (QUERY ANTI-PATTERN) NULL Usage
[Matching Expression: null]
```

How is this an anti-pattern? How else would I check for `NULL`?

On the other hand, comparisons such as `WHERE baz = NULL` are usually bugs and warrant a “high risk” message, I think.

Even your own [explanation for this “anti-pattern”](https://github.com/jarulraj/sqlcheck/blob/master/docs/query/3002.md) doesn’t explain what is wrong with it (or how it relates to the query at all, for that matter).

**Comment by jarulraj** [comment]
Fair point. I had hoped that these checks will simply provide hints for when to use `NULL` and `NOT NULL`. We could maybe add another check for queries that do this: `WHERE baz = NULL`.

**Comment by jarulraj** [comment]
Fixed this in 0bf8916  by adding an additional "hint" mode to not classify usage hints as anti-patterns.
